module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -64,48 +73,111 @@ namespace module::localisation {
+                                                           getFieldPosition(goal_post, fd, 1),  // Own post
+                                                           goals.Hcw);
+
+                                auto filter_new_opp = saved_filter;
+
+                                // Make a new candidate for the post
+                                auto candidate_opp =
+                                    filter_new_opp.measure(Eigen::Vector3d(m.position.cast<double>()),
+                                                           Eigen::Matrix3d(m.covariance.cast<double>()),
+                                                           getFieldPosition(goal_post, fd, 0),  // Opp post
+                                                           goals.Hcw);
+                                if (config.debug) {
+
+                                    Eigen::Vector3d state(filter.get());
+                                    Eigen::Transform<double, 3, Eigen::Affine> Hfw;
+                                    Hfw.translation() = Eigen::Matrix<double, 3, 1>(state.x(), state.y(), 0);
+                                    Hfw.linear() =
+                                        Eigen::AngleAxis<double>(state.z(), Eigen::Matrix<double, 3, 1>::UnitZ())
+                                            .toRotationMatrix();
+
+                                    const Eigen::Transform<double, 3, Eigen::Affine> Hcf(goals.Hcw
+                                                                                         * Hfw.inverse().matrix());
+
+                                    const Eigen::Matrix<double, 3, 1> rGCc_own(Hcf
+                                                                               * getFieldPosition(goal_post, fd, 1));
+                                    const Eigen::Matrix<double, 3, 1> rGCc_opp(Hcf
+                                                                               * getFieldPosition(goal_post, fd, 0));
+                                    log(fmt::format("Testing post {}",
+                                                    goal_post.side == Goal::Side::LEFT ? "left" : "Right"));
+                                    log(fmt::format("Candidate own {}", candidate_own));
+                                    log(fmt::format("State {}", state.transpose()));
+                                    log(fmt::format("Hcw {}\n", goals.Hcw));
+                                    log(fmt::format("Actual own post at {}",
+                                                    cartesianToSpherical(rGCc_own).transpose()));
+                                    log(fmt::format("Candidate opp {}", candidate_opp));
+                                    log(fmt::format("Actual opp post at {}",
+                                                    cartesianToSpherical(rGCc_opp).transpose()));
+                                    log(fmt::format("Measured post at {}", m.position.transpose()));
```

Comment:
If you're going to keep these logs, give them a log level, otherwise delete them